ai_genomics/utils/openalex.py

```diff
@@ -0,0 +1,140 @@
+# Functions to process OpenAlex data
+from itertools import chain
+from typing import Dict, List
+
+from pandas import DataFrame
+from toolz import pipe, partial
+
+
+def deinvert_abstract(inverted_abstract: Dict[str, List]) -> str:
+    """Convert inverted abstract into normal abstract
+
+    Args:
+        inverted_abstract: a dict where the keys are words
+        and the values lists of positions
+
+    Returns:
+        A str that reconstitutes the abstrat
+
+    """
+
+    abstr_empty = (max(chain(*inverted_abstract.values())) + 1) * [None]
+
+    for word, pos in inverted_abstract.items():
+        for p in pos:
+            abstr_empty[p] = word
+
+    return " ".join(abstr_empty)
+
+
+def get_institutions(doc: dict) -> list:
+    """
+    Create a list of all institutions involved in a work
+    """
+
+    return [
+        inst["id"]
+        for auth in doc["authorships"]
+        for inst in auth["institutions"]
+        if len(inst) > 0
+    ]
+
+
+def enrich_institutions(doc_inst: list, insts_dict: dict, variable: str):
```

Comment:
```suggestion
def enrich_institutions(doc_inst: list, insts_dict: dict, variable: str) -> List:
```

It's not immediately clear what the different args here are supposed to be. Maybe add `Args` to docstring? I also wonder whether it would be better to be consistent with terminology (i.e. `doc` -> `work`)